packages/jsapi-bootstrap/src/useWidget.ts

```diff
@@ -29,27 +30,50 @@ export function useWidget<T extends dh.Widget = dh.Widget>(
     widget: null,
     error: null,
   }));
-
+  const api = useApi();
+  const unsupportedTypes = useMemo(
+    () => [
+      api.VariableType.TABLE,
+      api.VariableType.TREETABLE,
+      api.VariableType.HIERARCHICALTABLE,
+      api.VariableType.TABLEMAP,
+      api.VariableType.PARTITIONEDTABLE,
+      api.VariableType.FIGURE,
+      api.VariableType.PANDAS,
+      api.VariableType.TREEMAP,
+    ],
```

Comment:
Is there any reason we couldn't/shouldn't support these? The `fetch` works the same, regardless of descriptor. It's just the returned type doesn't extend `JsWidget` so we would only close `exportedObjects` if that field exists.

I don't think we would call this hook with these types since they are all handled by other plugins right now, but I could see an argument for replacing the subscription code for `IrisGridPanel` and `ChartPanel` with this centralized listener code

Reply:
I think the main thing that would prevent us from using this hook in those panels is at least 1 requires a fetch function and not the widget I think (I can't remember which). IIRC it was something to do with reloading or resubscribing to the widget properly and the way the models interacted w/ the panel/component

Reply:
You might be thinking of charts built using `ChartBuilder`, I think that currently needs the fetch. Pretty sure I addressed that with my core plugins/xcomponent change if I recall correctly. 
My intent for this was useWidget is just for widgets. Could have another hook for `useTable`, `useFigure`, etc that is just built on top of `useObjectFetch`.
In any case, I wasn't going to replace the existing panels until Grizzly. This PR is intended for V+.

Reply:
I think you're right, it was chart builder.

And that's fine we can think about it at a later point if we add those hooks. One reason against it would be you can't call hooks conditionally (although you can with `use` in react 19 I think, but not sure on the specifics). So if you had a descriptor that could be any of the types, we wouldn't be able to fetch it with the hook. Or we would have to call multiple hooks and disregard the results of all but 1.

Reply:
Yeaaaa you know what, I think you're right - I'll just have it support all those widget types. Let the hook handle the messiness a little bit.